plugins/harness-ci-cd/src/hooks/useGetExecutionsList.ts

```diff
@@ -31,8 +34,10 @@ const useGetExecutionsList = ({
   const [currTableData, setCurrTableData] = useState<any[]>([]);
   const [totalElements, setTotalElements] = useState(50);
   const [flag, setFlag] = useState(false);
+  const identityApi = useApi(identityApiRef);
 
   useAsyncRetry(async () => {
+    const { token } = await identityApi.getCredentials();
```

Comment:
Hi! This fetches the Backstage user identity token. This is not required for making calls to the backstage proxy backend.

The code that this PR removes is only relevant when Backstage is running inside Harness, which is part of our [upcoming module - Internal Developer Portal](https://developer.harness.io/docs/internal-developer-portal/getting-started/overview#software-catalog). It should not impact the plugin running on self-hosted Backstage.

Reply:
Ah yes, you're correct. Re-reviewing the docs ([`Auth in Backstage`](https://backstage.io/docs/auth/#identity-for-plugin-developers) and [`Authenticate API Requests`](https://github.com/backstage/backstage/blob/master/contrib/docs/tutorials/authenticate-api-requests.md)) I see that it's only for backend to backend invocations or when the frontend plugin needs to call to an authentication locked backend. We use it on all of our requests internally so that way we don't have to add it in after-the-fact if something changed.

Looking at the errors here:
![image](https://github.com/harness/backstage-plugins/assets/1512028/102a4098-13af-4ec5-bde5-fb8db47dbecb)

I'm now realizing that it doesn't matter if we add the actual Backstage token or not. Where it's failing is trying to parse the JSON in the `getSecureHarnessKey` function, which in turn will have [it return undefined](https://github.com/harness/backstage-plugins/blob/main/plugins/harness-ci-cd/src/util/getHarnessToken.ts#L8). Which then [overrides the token value to be an empty string](https://github.com/harness/backstage-plugins/blob/main/plugins/harness-ci-cd/src/hooks/useGetExecutionsList.ts#L55). Which then [overrides the `Authentication` header with an empty string](https://github.com/harness/backstage-plugins/blob/main/plugins/harness-ci-cd/src/hooks/useGetExecutionsList.ts#L59).

So I guess by deduction we can assume that passing `Authorization: ""` to Harness then causes it to fail as a 401. That seems strange to me though considering the `x-api-key` is what's being set in the `app-config.yaml`.

Either way, this wasn't working until I modified the token to at least not pass the empty string. May be easier to just throw that `value` in as undefined instead of empty and call it a day.

Reply:
One more note, I'd still caution against throwing any kind of token in `localStorage`. It has several attack vectors which could cause problems for ya'all down the line. Not sure what your use-cases are, so don't want to throw out a bunch of suggestions on addressing it, but just wanted to point it out.

Also, since I'm here, [atob](https://github.com/harness/backstage-plugins/blob/main/plugins/harness-ci-cd/src/util/getHarnessToken.ts#L3) gets [flagged as `deprecated`](https://github.com/microsoft/TypeScript/issues/45566) in most TypeScript compilations. Just to avoid the IDE from flashing errors it may be better to replace it with `window.atob()` or `Buffer.from(data).toString('base64');` if you continue to use the `getHarnessToken` util.

Reply:
Two things here.
1. `getSecureHarnessKey` is something that is being used by harness only. For this to pass the plugin needs to work in Harness env. We should remove the error here as it is not relevant to the broader audience.
2. Getting 401 - If you are using`authMiddleware` then it is possible that `authMiddleware` is throwing an error. 401 from the harness is unlikely as the backstage proxy doesn't forward the `Authorization` header by default. So `Authorization: ""` should be causing 401 from `authMiddleware`.

According to what I understand the correction we should do is.
We add clarification around the use of `getSecureHarnessKey`. And by default, we send backstage token and for Harness specific use case we override the token.